### ISSUE TITLE
[Feature request] More keywords support for `not` keyword 

### ISSUE THREAD

**Issue by ehmicky** [json-schema-faker/json-schema-faker#411]
At the moment, the following keywords are supported as a child to `not`: `type`, `[exclusive]minimum|maximum`, `minLength`, `maxLength`, `required`.

It would be great to also support the following keywords: `format`, `multipleOf`, `pattern`.

Although less likely, technically the following keywords are also valid keywords under `not`: `items`, `additionalItems`, `minItems`, `maxItems`, `minProperties`, `maxProperties`, `properties`, `additionalProperties`, `patternProperties`, `uniqueItems`

**Comment by ehmicky** [comment]
Yes I guessed this was a hard one!

My primary concern is that without it, it is possible to generate mock data from `json-schema-faker` that actually fails validation (using library like `ajv`) with the same JSON schema. Since I use `json-schema-faker` to generate automated tests, this means my tests might randomly fail.

I think the problem is hard for two reasons:
  - `not` means returning _anything_ that does not match specific constraints. While specific keywords are difficult, as soon as you had `allOf` and the other combinators, that probably becomes even harder. 
  - this _anything_ must still match the parent constraints. This is for example the reason for the bug #410.

Several proposed solutions:
  - adding a bunch of `if` and hard-coded logic for each keyword. That's the current solution. It's straight-forwarded but might become complex as more keywords are combined together.
  - defining each keyword as a function that takes a single parameter and return `true` or `false` depending on whether the constraint passes. Then run a bunch of iterations until a value is found. Several downsides: it is very slow and inefficient. It also will run even if no value can be logically found.
  - defining each keyword sort kind of "set" from a mathematical standpoint, and somehow using intersections of those sets. A little like purely functional languages might do, or like libraries like [QuickCheck](https://hackage.haskell.org/package/QuickCheck). This might be the best solution, but also I have no idea if this is even possible!

**Comment by pateketrueke** [comment]
Yeah, this is a _pretty_ one... at some point we need to really validate the generated input, so jsf would become eventually a validator itself?

I discouraged that idea from the beginning... the truth, is that generating something very random _will always satisfy_ a `not` schema? You can simple provide a `not` with two very big and complex schemas with dependencies from each other, and generating a value could be as trivial as generating an empty string, or just a -1.

Like in RegExp sometimes is more clear to negate things rather than providing a pattern to match.

There's nothing more accurate the being specific and getting exactly what you need...

I really want a valid `not` generator!
